p2p/ipld/write_test.go

```diff
@@ -2,196 +2,156 @@ package ipld
-	}
-}
-func generateRandTxs(num int, size int) [][]byte {
-	randMsgs := generateRandNamespacedRawData(num, nmt.DefaultNamespaceIDLen, size)
-	for _, msg := range randMsgs {
-		copy(msg[:nmt.DefaultNamespaceIDLen], consts.TxNamespaceID)
-	}
-	return randMsgs
-}
-
-func toMessageSlice(msgs [][]byte) []*tmproto.Message {
-	res := make([]*tmproto.Message, len(msgs))
-	for i := 0; i < len(msgs); i++ {
-		res[i] = &tmproto.Message{NamespaceId: msgs[i][:nmt.DefaultNamespaceIDLen], Data: msgs[i][nmt.DefaultNamespaceIDLen:]}
-	}
-	return res
-}
-
-func TestDataAvailabilityHeaderRewriteBug(t *testing.T) {
-	logger := log.TestingLogger()
-	dag := mdutils.Mock()
-	croute := ipfs.MockRouting()
-
-	txs := types.Txs{}
-	l := len(txs)
-	bzs := make([][]byte, l)
-	for i := 0; i < l; i++ {
-		bzs[i] = txs[i]
-	}
-	app := &preprocessingApp{}
-
-	// See state.CreateProposalBlock to understand why we do this here:
-	processedBlockTxs := app.PreprocessTxs(abci.RequestPreprocessTxs{Txs: bzs})
-	ppt := processedBlockTxs.GetTxs()
-
-	pbmessages := processedBlockTxs.GetMessages()
-
-	lp := len(ppt)
-	processedTxs := make(types.Txs, lp)
-	if lp > 0 {
-		for i := 0; i < l; i++ {
-			processedTxs[i] = ppt[i]
-		}
-	}
-
-	messages := types.MessagesFromProto(pbmessages)
-	lastID := makeBlockIDRandom()
-	h := int64(3)
-
-	voteSet, _, vals := randVoteSet(h-1, 1, tmproto.PrecommitType, 10, 1)
-	commit, err := types.MakeCommit(lastID, h-1, 1, voteSet, vals, time.Now())
-	assert.NoError(t, err)
-	block := types.MakeBlock(1, processedTxs, nil, nil, messages, commit)
-	block.Hash()
-
-	hash1 := block.DataAvailabilityHeader.Hash()
-
-	ctx := context.TODO()
-	err = PutBlock(ctx, dag, block, croute, logger)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	block.Hash()
-	hash2 := block.DataAvailabilityHeader.Hash()
-	assert.Equal(t, hash1, hash2)
-
-}
-
-func generateRandomMsgOnlyData(msgCount int) types.Data {
-	out := make([]types.Message, msgCount)
-	for i, msg := range generateRandNamespacedRawData(msgCount, consts.NamespaceSize, consts.MsgShareSize-2) {
-		out[i] = types.Message{NamespaceID: msg[:consts.NamespaceSize], Data: msg[consts.NamespaceSize:]}
-	}
-	return types.Data{
-		Messages: types.Messages{MessagesList: out},
-	}
-}
-
-func makeBlockIDRandom() types.BlockID {
-	var (
-		blockHash   = make([]byte, tmhash.Size)
-		partSetHash = make([]byte, tmhash.Size)
-	)
-	mrand.Read(blockHash)
-	mrand.Read(partSetHash)
-	return types.BlockID{
-		Hash: blockHash,
-		PartSetHeader: types.PartSetHeader{
-			Total: 123,
-			Hash:  partSetHash,
-		},
-	}
-}
-
-func randVoteSet(
-	height int64,
-	round int32,
-	signedMsgType tmproto.SignedMsgType,
-	numValidators int,
-	votingPower int64,
-) (*types.VoteSet, *types.ValidatorSet, []types.PrivValidator) {
-	valSet, privValidators := types.RandValidatorSet(numValidators, votingPower)
-	return types.NewVoteSet("test_chain_id", height, round, signedMsgType, valSet), valSet, privValidators
-}
+// TODO(Wondertan): Find a proper place for this test
```

Comment:
Are we planning on handling this before this PR is merged or after?

Reply:
I want to do this before merge surely, but I don't know exactly where to move this test. What I know is that `p2p/ipld` shouldn't depend on ABCI and maybe there is also a way to simulate this bug without depending on it, or we could simply move the test to `types` and no more worry about it. Thoughts?